docs/website/docs/building-from-source/getting-started.md

```diff
@@ -138,32 +144,102 @@ cmake --build ../iree-build/
 
     See also our [developer documentation for ccache](https://github.com/openxla/iree/blob/main/docs/developers/developing_iree/ccache.md).
 
-## What's next?
+### Optional components
+
+By default, the CMake build includes:
+
+* All compiler targets (`llvm-cpu`, `cuda`, `vulkan-spirv`, etc.)
+* All runtime HAL drivers (`local-task`, `cuda`, `vulkan`, etc.)
+* All compiler input formats (StableHLO, TOSA, etc.)
+* All compiler output formats (VM bytecode, C)
 
-<!-- TODO(scotttodd): "at this point you can..." -->
+The default build does _not_ include:
+
+* Compiler or runtime bindings (Python, TFLite, etc.)
+* Advanced features like AddressSanitizer or tracing instrumentation
+* Experimental components
+
+These can be changed via the `IREE_` CMake options listed in the root
+[`CMakeLists.txt`](https://github.com/openxla/iree/blob/main/CMakeLists.txt).
+
+### Extensions and project integrations
+
+When using IREE within other projects, you can register compiler plugins and
+runtime HAL drivers. You can also bring your own copy of LLVM and some other
+tools. See the root
+[`CMakeLists.txt`](https://github.com/openxla/iree/blob/main/CMakeLists.txt)
+for details.
+
+## Tests and samples
 
 ### Running tests
 
-Build test dependencies and run tests:
+Tests are run via [ctest](https://cmake.org/cmake/help/latest/manual/ctest.1.html).
+To build and run the core project tests:
 
 ``` shell
-cmake --build ../iree-build --target iree-run-tests
+# Build default targets
+cmake --build ../iree-build/
+
+# Run tests
+ctest --test-dir ../iree-build/
 ```
 
-Internally, this builds dependencies via the `iree-test-deps` target and
-invokes [CTest](https://gitlab.kitware.com/cmake/community/-/wikis/doc/ctest/Testing-With-CTest).
+!!! Caution
+    This has two limitations:
+
+    1. Large tests are excluded from the build by default
+    2. Some tests require hardware like a GPU and will fail on unsupported systems
 
-The parallel testing level can be set via the environment variable
-`CTEST_PARALLEL_LEVEL` when invoking ctest in this fashion. Instructions
-are printed to test with a custom command line.
+To build and then run all tests:
 
-### Take a look around
+``` shell
+# 1. Build default targets
+cmake --build ../iree-build/
 
-Check out the contents of the 'tools' build directory:
+# 2. Build test dependencies
+cmake --build ../iree-build/ --target iree-test-deps
+
+# 3. Run tests
+ctest --test-dir ../iree-build/
+
+
+# Or combine all steps using a utility target
+cmake --build ../iree-build --target iree-run-tests
+```
+
+To run only certain tests, we have a
+[helper script](https://github.com/openxla/iree/blob/main/build_tools/cmake/ctest_all.sh)
+that converts environment variables into ctest filters:
```

Comment:
Should we mention other scripts like `build_all.sh` too?

Reply:
These scripts are really designed for CI (and say so at the top). They were made possible to use locally for the purpose of keeping CI reproducible, not for providing an ergonomic entry point for developers. ctest_all.sh is mostly pretty usable as a development tool, but especially the build scripts were really not aimed at that. I'd prefer we not direct users to them and create expectations that they are something other than what they are.